### ISSUE TITLE
False-positive "Argument 1 to np.concatenate has incompatible type"

### ISSUE THREAD

**Issue by bersbersbers** [python/mypy#12144]
**Bug Report**

False-positive "Argument 1 to np.concatenate has incompatible type"

**To Reproduce**

```python
import numpy as np
print(np.concatenate((np.array([1]), [2])))
```

**Expected Behavior**

`python bug.py` works fine, printing `[1 2]`, so I would expect `mypy bug.py` not to report an error.

**Actual Behavior**

> Argument 1 to "concatenate" has incompatible type "Tuple[ndarray[Any, dtype[Any]], List[int]]"; expected "Union[_SupportsArray[dtype[<nothing>]], Sequence[_SupportsArray[dtype[<nothing>]]], Sequence[Sequence[_SupportsArray[dtype[<nothing>]]]], Sequence[Sequence[Sequence[_SupportsArray[dtype[<nothing>]]]]], Sequence[Sequence[Sequence[Sequence[_SupportsArray[dtype[<nothing>]]]]]]]"

**Your Environment**

- Mypy version used: mypy 0.931
- Mypy command-line flags: none
- Mypy configuration options from `mypy.ini` (and other config files): none
- Python version used: 3.10.2
- Operating system and version: Linux

**Comment by JMMarchant** [comment]
This is related to https://github.com/numpy/numpy/issues/20901 and appears to be an `@overload` issue of some description.

I can confirm this still exists on numpy 1.22.4 and mypy 1.1.1
